### PR TITLE
STORM-3437: Automate license checks more

### PR DESCRIPTION
**Pull request by srdo**
Based on https://github.com/apache/storm/pull/3051.

The idea here is to add license checks to CI, so PRs fail unless they keep the LICENSE-binary and DEPENDENCY-LICENSES files up to date.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,6 +40,12 @@ before_install:
   - sudo apt-get update
   - sudo apt-get install python3
   - sudo apt-get install python3-pip
+  - sudo add-apt-repository ppa:jonathonf/python-3.6 -y
```
[the remark this answers is not in the document]
Comment (srdo) [reply .travis.yml]: The Travis image is using an older Python version than 3.6. I think the script needed something added in 3.6, but I forget what. I'll try removing these lines and seeing whether the Travis build fails.

File: .travis.yml [hunk 2]
```diff
@@ -39,6 +40,12 @@ before_install:
...
+  - sudo apt-get update
+  - sudo apt-get install python3.6
+  - wget http://mirrors.rackhosting.com/apache/maven/maven-3/3.6.1/binaries/apache-maven-3.6.1-bin.tar.gz -P $HOME
```
[the remark this answers is not in the document]
Comment (srdo) [reply .travis.yml]: The Travis image is using an old Maven version, and the license plugin needs a newer Maven version

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ethanlm) [reply .travis.yml]: If we need python3.6, maybe we can do
``` 
sudo apt update
sudo apt install python3.6
```
I am just concerned about pulling a personal? repo

[the remark this answers is not in the document]
Comment (srdo) [reply .travis.yml]: Python 3.6 isn't packaged for Ubuntu 14.04. We either need to switch to Xenial (not sure whether this is easy, it seems like at minimum we'd need to get Java 8 from somewhere else https://travis-ci.community/t/install-of-oracle-jdk-8-failing/3038/8), or get it from a PPA.

I'm happy to use https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa instead?

[the remark this answers is not in the document]
Comment (Ethanlm) [reply .travis.yml]: I am not sure how they maintain ppa and if we can trust them. But I am fine with it if it's the only option we have as of now.

[the remark this answers is not in the document]
Comment (srdo) [reply .travis.yml]: The script was using string interpolation. Going to try removing it and seeing if that was the only use of new features.

[the remark this answers is not in the document]
Comment (srdo) [reply .travis.yml]: Okay, it wasn't. I also use expanduser from Python 3.5. I'll switch back to installing Python 3.6 and use the deadsnakes ppa.

[the remark this answers is not in the document]
Comment (Ethanlm) [reply .travis.yml]: Thanks for trying it

**Comment by srdo** [comment]
Thanks for reviewing.

**Pull request merged**